Font.js

```diff
@@ -84,9 +84,8 @@ function getFontCSSFormat(path) {
         ttc: `Based on the current CSS specification, font collections are not (yet?) supported.`
     }[ext];
 
-    if (!msg) msg = `${url} is not a font.`;
-
-    this.dispatch(new Event(`error`, {}, msg));
+    if (!msg) msg = `'${path}' is not a font.`;
+    throw new Error(msg);
```

Comment:
Good catch on the `${url}`, but let's keep this the `this.dispatch()` call, because this should end up triggering error handling via events, rather than crash the script.

Reply:
though the event manager should probably be enriched a little so that if the event.type is `error` or `load`, and the instance has an `onerror` or `onload` property defined, that function gets auto-triggered in addition to running through the list of registered event handling functions.

Reply:
We could keep the `this.dispatch`, but then this function will need to be moved from top level to a private function in the Font class.

I think throwing the exception here is pretty elegant btw, it's automatically handled by the code that's in place here in `loadFont`: https://github.com/Pomax/Font.js/pull/95/files#diff-66361256e5eedd645ac47a4b5a5ca7526edaa1e162503a61c5be6c0cc902aff8R165. See also my explanation in my other comment.

Reply:
That's one feature of JS I will refuse to ever work with: a large part of the power of JS is that _everything_ stays accessible, and people can explore and monkey-patch to their heart's content just by firing up dev tools. Adding private properties makes the web that little bit worse and I won't be a party to it =)

Having said that, the reason I want to steer clear of throwing errors here is that technically this function "can't" error out: even if it gets a resource that isn't a font as far as it can tell, there is no unexpected code situation we need to signal a runtime error for: this case is entirely expected. To make things more interesting: this code doesn't even _need_ to succeed. It will fail to build a `@font-face` stylesheet, but that is in no way required for `loadFont` to then do its job. We need to wait for it, in case it succeeds, because if it does loadFont will be conveniently hitting the cache, but failing _is_ a perfectly valid option in this case.

What I think I'm going to do is open a new issue to outline a better way to get optional `@font-face` injection to happen as an awaitable `async` call. I should be able to get that turned into a PR today and will tag you on that to see if that's a workable solution to the same problem, but with more control for the user.